components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

```diff
@@ -146,7 +146,11 @@ fun String.sanitizeURL(): String {
  * For example for an input of "/../../../../../../directory/file.txt" you will get "file.txt"
  */
 fun String.sanitizeFileName(): String {
-    return this.substringAfterLast(File.separatorChar)
+    val file = File(this.substringAfterLast(File.separatorChar))
+    // Remove unwanted dots in the file name.
+    val name = file.nameWithoutExtension.replace(".", "")
```

Comment:
This will work but I am wondering if we should support file names like `test.2020.12.01.txt`. They would currently come out as `test20201201.txt`. I assume only subsequent "." are a problem, not multiple indiviudal "." in a name which should be fine.

If we want to support this you could change this to just:
```
val file = File(this.substringAfterLast(File.separatorChar))
return file.name.replace("\\.\\.+".toRegex(), ".")
```

To replace 2 or more "." with one, instead of clearing them all.

Reply:
Thanks for the feedback, sounds fair, I will do some test to see if it doesn't crash :)

Reply:
Confirmed the issue is only with subsequent dots, I will add this part!

Reply:
I added some additional checks, for cases like `file.` desktop returns `file`, also hidden files `.file` desktop returns `file` and I added the `sanitizeFileName` to `onExternalResponse` on the engine session to have the first prompt to display right file name, from a input of `test.......txt` it displays:
 
<img width="399" alt="image" src="https://user-images.githubusercontent.com/773158/100789028-e3f4dc00-33e3-11eb-8c7f-ad77d909caf9.png">

Reply:
I also file https://github.com/mozilla-mobile/android-components/issues/9085 as a discovered that `DownloadUtils.guessFileName` is not working properly for hidden files `.file`, let me know if you want to take a second look on the pr :)